sdk/core/core/inc/az_http_request_builder.h

```diff
@@ -77,7 +77,8 @@ AZ_NODISCARD az_result az_http_request_builder_init(
     az_mut_span const buffer,
     uint16_t const max_url_size,
     az_span const method_verb,
-    az_span const initial_url);
+    az_span const initial_url,
+    az_span const body);
```

Comment:
It might be nice to introduce a constant global az_span_null variable whose pointer is NULL and size is 0 which can be passed in when there is no body it other methods when there is no span to pass.

Reply:
We consider that this span `{ .begin = NULL, .size = 0 }` is an empty span (like an empty string in other languages. No span or null span should be a different value.

Reply:
I'm ok to have a static const `az_span_empty` or a function `az_span az_span_create_empty() { return (az_span){ .begin = NULL, .size = 0 }; }` and use this value to indicate that there is no body. I just would like to clarify that we need another value if empty span has different meaning compare to no/null span.

Reply:
`(az_span){ 0 }` is az_span_null, in my opinion